### PR TITLE
Fishing map/fix empty env layers

### PR DESCRIPTION
**Pull request by j8seangel**
Fixes style validation and black environmental layers with NaN tooltips:

image

### CODE REVIEW

File: libs/fourwings-aggregate/package.json [hunk 1]
```diff
@@ -2,7 +2,9 @@
   "name": "@globalfishingwatch/fourwings-aggregate",
   "version": "4.1.0",
   "description": "Logic to turn fourwings tiles or cells into meaningful values for the frontend",
-  "main": "dist/index.js",
+  "main": "../../dist/libs/fourwings-aggregate/index.umd.js",
```
Comment (j8seangel) [inline libs/fourwings-aggregate/package.json:5]: useful to get fourwings-aggregate link with maplibre working

File: apps/fishing-map/features/analysis/analysis.hooks.ts [hunk 2]
```diff
@@ -120,7 +121,18 @@ export const useFilteredTimeSeries = () => {
       setBlur(true)
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [timeComparison, timebarStart, timebarEnd])
+  }, [timeComparison])
+
+  useEffect(() => {
```
Comment (j8seangel) [inline apps/fishing-map/features/analysis/analysis.hooks.ts:126]: it used to add a blur effect for environmental layers when not needed and never removed

File: libs/ocean-areas/project.json [hunk 3]
```diff
@@ -37,6 +37,12 @@
         "jestConfig": "libs/ocean-areas/jest.config.js",
         "passWithNoTests": true
       }
+    },
+    "deploy": {
```
Comment (j8seangel) [inline libs/ocean-areas/project.json:41]: now just run `nx deploy [package-name]` to publish in npm registry

File: libs/react-hooks/src/use-layer-composer/use-layer-composer.ts [hunk 4]
```diff
@@ -45,11 +45,15 @@ export function useLayerComposer(
         )
         const afterTransformations = applyStyleTransformations(style, styleTransformations)
         if (process.env.NODE_ENV === 'development') {
-          if (!styleSpec) {
-            styleSpec = await import('@globalfishingwatch/maplibre-gl-style-spec')
+          if (!styleSpecValidate) {
```
Comment (j8seangel) [inline libs/react-hooks/src/use-layer-composer/use-layer-composer.ts:48]: fixes import and enable the style validation with errors help again

File: apps/fishing-map/features/timebar/TimebarActivityGraph.utils.ts [hunk 5]
```diff
@@ -29,12 +36,23 @@ export const getChunksTimeseries = ({
...
             return [key, cleanValue]
           })
-        )
+        ) as TimeSeriesFrame
         return {
           ...activeFrameValues,
           date: quantizeOffsetToDate(frameValues.frame, interval).getTime(),
         }
       })
+      if (aggregationOperation === AggregationOperation.Avg) {
```
Comment (nerik) [inline apps/fishing-map/features/timebar/TimebarActivityGraph.utils.ts:45]: Thanks 👍 
Making a note to check in agg-tile why exactly this is happening and implement a test

**Review by nerik — COMMENTED** [review]
Thanks, looks good.
I have fixed in a commit in this branch the bug I introduced initially with the month interval in env layers https://github.com/GlobalFishingWatch/frontend/pull/1272/commits/965d2200fd69c449212abe1af3676dcabc6212b0

File: libs/timebar/package.json [hunk 6]
```diff
@@ -31,7 +31,7 @@
   },
   "devDependencies": {
     "@mapbox/geojsonhint": "^3.0.0",
-    "@globalfishingwatch/maplibre-gl-style-spec": "^14.0.2",
+    "@globalfishingwatch/maplibre-gl": "^2.1.6-gfw.13",
```
[the remark this answers is not in the document]
Comment (j8seangel) [reply libs/timebar/package.json:34]: Originally it is released as a separated package for the cli usage and not included in the library
But as we don't use the cli we can use directly from the library for simplicity.

**Note**: I needed to add it in the dist folder see https://github.com/GlobalFishingWatch/maplibre-gl-js/pull/2/commits/e2ffe41c1ec0381a839b3bd94218248cf5d2e0a6

File: libs/react-hooks/src/use-map-interaction/use-map-interaction.ts [hunk 7]
```diff
@@ -104,7 +104,8 @@ const getExtendedFeatures = (
           console.log(properties.rawValues)
         }
 
-        if (!values || !values.filter((v: number) => v !== 0).length) return []
+        // Clean values with 0 for sum aggregation and with NaN for avg aggregation layers
+        if (!values || !values.filter((v: number) => v !== 0 && !isNaN(v)).length) return []
```
[the remark this answers is not in the document]
Comment (j8seangel) [reply libs/react-hooks/src/use-map-interaction/use-map-interaction.ts:108]: Probably not needed anymore, it was a previous fix

**Pull request merged**